Xamarin.Forms.Core/Application.cs

```diff
@@ -314,6 +317,7 @@ void OnPopCanceled()
 			if (_saveAgain)
 				await DependencyService.Get<IDeserializer>().SerializePropertiesAsync(Properties);
 			_isSaving = _saveAgain = false;
+			SaveSemaphore.Release();
```

Comment:
This should be in a `try...finally` block so we can guarantee that the semaphore is released if there's an exception while saving the properties.

Reply:
should I catch and log the exception similar to what's done in Xamarin.Forms.Platform.Android.Deserializer.DeserializePropertiesAsync()?